Minimalist Theme

konero opened the pull request:
# Overview
A theme built around modules and a single wireframe, so multiple theme colors can be output and changes made to any of the layout files such as adding new features only need to be done once, not three times.

Written in **Less**.

# Todo
- [x] Gray-048
- [x] Gray-072 (default)
- [x] Gray-128
- [x] Convert all QSS icons to SVG

# Design
The original themes were used for the palettes. I wanted to make it look minimal so people who used it for painting can find missed pixels without distraction. It was also important to me to retain as much of the legacy as possible. I also applied a bunch of accessibility standards.

## Preview 048 Very Dark
![new-48-prev](https://cloud.githubusercontent.com/assets/19820721/24840691/6d773df0-1d69-11e7-817d-c144b81632b0.png)

## Preview 072 Dark
![new-072-prev](https://cloud.githubusercontent.com/assets/19820721/24840582/998b9550-1d67-11e7-9ff0-f9f2996e4548.png)

## Preview 128 Neutral
![new-128-prev](https://cloud.githubusercontent.com/assets/19820721/24840678/323b9b64-1d69-11e7-89f1-9b3f771cd9d6.png)

# How to Develop
Base | Components | Layouts
----- | -------------- | -------
Contains commonly used blocks and variables across all variants. | Contains reusable modules such as button styles, tab structures and icons. | Contains the wireframe for the project, each widget is split into a separate file. | Contains small overrides. Works similarly to components.

## Using Components
Keep the cascade in mind in the table above, anything to the right can override anything to the left. So for instance if we develop a new feature and extend to a component, we can override that component in the same block.

````less
// component file
#component {
    color: @palette-red;
    padding: 10px;
}

// layout file
#feature {
    &:extend(#component all);
    color: @palette-green;
}
````

Will compile to...

````css
/* qss compiled */
#component,
#feature {
    color: red;
    padding: 10px;
}
#feature {
    color: green;
}
````

### Notes
* When adding new variables - add them to the default theme first (072/Dark). This is important because any alternate themes will have a dependency on it, the way alternate themes work is by overriding the necessary default theme variables.

* Never hard-code color values into the layout files, use variables. Layouts are meant to be wireframes - colors should be controlled from a theme file.

## Creating Icons
Icons require two variants for dark and light theme support, **black** and **white**.

Black | White
------ | ------
Is for light themes. | Is for dark themes.

To place a new img into a layout file use the `@img-url` variable and assume you are at the root of both **imgs/black** and **imgs/white** - which one is true is dependent on the theme file.

For it to work in URLs you need to use [variable interpolation](http://lesscss.org/features/#variables-feature-variable-interpolation) (`@{img-url}`).

````less
#feature {
    image: url('@{img-url}/new_icon.png');
}
````

## Code Style Guide
* **Tab Indent:** 2 Spaces.
* **Ordered**: Alphabetically.

# How to Compile
To compile the default theme, compile **main.less**. To compile an alternate theme color `@import` **main.less** into the theme file and compile the theme file.

### My Compile Setup
I use [WinLess](http://winless.org/). Make sure **minify** is unchecked. With WinLess, any file I edit and save will recompile all theme files automatically.

![comp-wl](https://cloud.githubusercontent.com/assets/19820721/24840975/12708c42-1d71-11e7-8016-c928e130ca11.PNG)

# Features
* Scrub bar has been redesigned.
* FPS slider has been made to respond to small windows, it can fall back to buttons only.
* Console bars are now all equal in height for side-by-side docking.
* Input boxes now have focus color and selection color.
* Menus display which items are check-able, which tool is selected and disabled items now _feel_ disabled for Windows and Linux builds.
* Drop down menu is now styled.
* Treeview in various places now use correct branch sizes.
* Changed small lock icon from Camera Settings to a radio button.
* Alternate background colors were removed on treeviews that display no context.
* Titlebars increased in height for extra text margin.
* Titlebars that display no text now display a grip bar.
* Increased hit area for QSlider, tool buttons and QScrollBar handle.

**Scrub Bar**
Now displays the rendering frame more clearly.

![e44188e8-10f6-11e7-9afc-f74ccbed3d84](https://cloud.githubusercontent.com/assets/19820721/24346708/7bfbff2e-12cd-11e7-9796-c9e44a2bfaf8.gif)

**FPS Slider**
Slider responds to shrinkage.

![responsive-fps-slider](https://cloud.githubusercontent.com/assets/19820721/24346666/41352f82-12cd-11e7-8ab8-4b2a9993ab88.gif)

# Bugs
These are beyond the scope of this PR. This PR is based on QSS/LESS only, not C++.
* DoublePairField (double QSlider) needs to be reimplimented.
* Accent lines at the top of the Console Play Bar in ComboViewer doesn't show when the window has been opened from the Menu. The app must be relaunced or the theme changed for it to show.
* Input boxes in various places are being clipped.
* The font config could be cleaned so MS Windows uses Segoe UI instead of Arial. This can't be changed in the QSS because it causes conflicts with Linux.

## Notes
I haven't tested this on Mac, due to feedback from @artisteacher in the previous PR I have rolled back any font configuration until that is sorted out in the app itself.

# Download RELEASE
You can download the following zip file and place the folders inside in your stuffs folder in **config/qss**.

**Updated Apr/10/2017**
### [Astral_ThemePack_Apr10_2017.zip](https://github.com/opentoonz/opentoonz/files/911482/Astral_ThemePack_Apr10_2017.zip)

turtletooth commented:
Nice work.  One quick thing I found:
When the theme has already been loaded, closing and reopening OpenToonz and going to preferences and clicking a new section results in this:

![theme_error](https://cloud.githubusercontent.com/assets/4576381/24121367/a455263a-0d7d-11e7-9d82-73e68794486f.PNG)

konero commented:
@turtletooth Thanks! Wouldn't look nearly half as clean without the countless fixes you merged already.

Is that a consistent thing that happens with the preferences window? I noticed that exact bug once when I was doing the Linux tests but I couldn't re-produce it.

I've re-implemented the Preferences Window style, could you give this a try?

[Minimalist_072.zip](https://github.com/opentoonz/opentoonz/files/856675/Minimalist_072.zip)

tGuMedia commented:
looking good, i'm surpriced you didn't add the viewer icons you made.

adamearle commented:
Really really looks fantastic mate. Honestly really well done. Actually
what did happen to your icons?

Adam Earle





On 21 Mar 2017 07:06, "Kite" <notifications@github.com> wrote:

Overview

A theme built around modules and a single wireframe, so multiple themes can
be output from it and changes made to any of the layouts such as adding new
features only need to be done once.
Note on Design

The original 072 theme was used for the color palette. I wanted to make it
as minimal as I could get it to be so people who used it for painting can
find missed pixels without the UI distracting their eye.

I applied accessibility standards, such as passing the WCAG 2.0
<https://www.w3.org/TR/WCAG20/> 4.5:1 contrast test.
Preview (Windows 10)

[image: win10]
<https://cloud.githubusercontent.com/assets/19820721/24118725/b7ec726a-0da6-11e7-8464-e6536d1455a4.png>
Preview (Linux Mint 18)

[image: mint]
<https://cloud.githubusercontent.com/assets/19820721/24118730/bd55f730-0da6-11e7-87cd-3f4a903f15be.png>
How to Develop

*A Word on Components*
Since this theme is built around the idea of components (or modules) keep
this in mind when developing. For a list of components, open the components/
folder.

If I remember correctly with the original themes, tabs were styled 3 or 4
times in a single file, so if we wanted to make a change across all 3
themes that would be 9-12 changes. Instead we can just do this:

&:extend(.tab-flat all);

And if we wanted to make a change, we make the change in the component
file, which will update the style anywhere the component is being extended
from.

A more complicated example would be this, we can add a button style to a
new widget and also an icon, purely by using components.

&:extend(.button-tool all);
&:extend(.icon-lock all);

Which applies the tool button style and the lock icon, including all their
hover, pressed, checked and disabled states etc.

*Notes*

   - Always make sure when developing that if you add new variables to add
   them to the default theme file first; this is located in
   themes/Minimalist_072.less. This is important because any alternate
   themes will have a dependency on it and will use it as a fall back if they
   don't have that new variable.
   - Never hard-code color values into the layouts/ files, remember these
   are only wireframes and should only contain variables for color values.

Style Guide

   - *Tab Indent:* 2 Spaces.
   - *Ordered*: Alphabetically.

How to Compile

Compile the main file called main.less located in less/ and make sure the
output filename matches the folder name of the theme located in ../qss.
Custom Themes

Requires a LESS compiler that supports @import.

For custom themes, create a new file in less/themes and @import the main
file located in less/.

@import '../main';

Open the default theme file that contains all the default variables located
in /themes and copy/paste only the variables you wish to override into the
custom theme file. To compile the custom theme, you compile the custom
theme file. Make sure to create a new folder in the /qss folder with the
name of the custom theme QSS file.
Features

   - Input boxes now have focus.
   - Input boxes now has text-selection style.
   - Menus display which items are checkable. (Win/Linux)
   - Menus display more clearly which tool is selected. (Win/Linux)
   - Menus disabled items now "feel" disabled by removing their hover
   states.
   - The Combo Box menu is now styled.
   - The tree view in Studio Palette now uses correct branch size.
   - The small lock icon under Camera Settings is now a radio button.
   - Play bars are now all equal in height.
   - Alternate background colors were removed on tree views that display no
   contiguous information in their columns.
   - Tree and browse views now have a different background color to the
   main panel.
   - Title Bars have been increased in height to give text some extra
   margin.
   - Title Bars that display no text now display a grip handle.
   - Increased hit area for QSlider control, Console Play Bar buttons, and
   QScrollBar handle; it's now grabbable even if the cursor goes "off screen".

Bugs (beyond the scope of this PR)

   - DoublePairField (double QSlider) needs to be reimplimented.
   - Accent lines at the top of the Console Play Bar in ComboViewer doesn't
   show when the window has been opened from the Menu. The app must be
   relaunced for it to show.
   - Input boxes in various places are being clipped.
   - IMO the font config could be cleaned so Windows uses Segoe UI instead
   of Arial. This can't be changed in the QSS because it causes conflicts with
   Linux.

Mac Notes

I haven't tested this on Mac, due to feedback from @artisteacher
<https://github.com/artisteacher> in the previous PR I have rolled back any
font configuration until that is sorted out in the app itself.
Linux Support

It now supports Linux (tested Ubuntu 16 and Mint 18). I had to rework a lot
of the cascade and roll some stuff back, but many people (Mint users) were
enthusiastically pressing for support, I did a line-by-line refactor with
Linux in mind.

   -

   Note: There is an underlying issue with fonts in the app itself. When
   the theme is loaded the first time it's best to restart OpenToonz and let
   the theme load at launch.
   -

   Note: Linux users for the time being can edit the QSS file and add this
   at the top to change their font preference. It's basic CSS, so very easy to
   look up. For font-weight, 400 = regular and 700 = bold.

QWidget {
  font-family: Sans; /* or 'Ubuntu' */
  font-size: 13px; /* or 12px */
  /* font-weight: 400; */
}

Comment

This was previouly submitted as a PR, sorry about that. I made a new PR
because they are quite different now, it required a line-by-line refactor
and the cascade has changed.

Feedback is welcome. Please keep in mind this PR's scope is based on QSS
only, but feel free to add suggestions.

@adamearle <https://github.com/adamearle>, @turtletooth
<https://github.com/turtletooth>, @shun-iwasawa
<https://github.com/shun-iwasawa>
Download

If you would like to use it, you can download the following zip file and
place it in your stuffs folder in *config/qss*.

*Minimalist_072.zip
<https://github.com/opentoonz/opentoonz/files/856210/Minimalist_072.zip>*
------------------------------
You can view, comment on, or merge this pull request online at:

  https://github.com/opentoonz/opentoonz/pull/1112
Commit Summary

   - Minimalist theme default

File Changes

   - *A* stuff/config/qss/Minimalist_072/Minimalist_072.qss
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-0> (1628)
   - *A* stuff/config/qss/Minimalist_072/Minimalist_072_mac.qss
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-1> (1628)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/arrow_down.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-2> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/arrow_left.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-3> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/arrow_right.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-4> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/arrow_up.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-5> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/cam_lock.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-6> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/cam_unlock.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-7> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/check_
   indicator-disabled.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-8> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/check_indicator.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-9> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/colorslider_add.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-10> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/colorslider_sub.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-11> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/combo_down_
   arrow-disabled.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-12> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/combo_down_arrow.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-13> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/flipmarker.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-14> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/flipslider.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-15> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/fsp_checked.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-16> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/fsp_unchecked.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-17> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/gear.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-18> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/lineedit_shadow.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-19> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/minus.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-20> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/plus.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-21> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/radiobutton_checked.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-22> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/radiobutton_
   unchecked.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-23> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/scroll-down.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-24> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/scroll-left.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-25> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/scroll-right.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-26> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/scroll-up.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-27> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/segment_disabled.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-28> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/segment_linked.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-29> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/segment_unlinked.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-30> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/selectiontool_
   position.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-31> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/selectiontool_
   rotation.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-32> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/selectiontool_
   scalepeg.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-33> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/selectiontool_
   thickness.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-34> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/slider_handle.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-35> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/slider_left_handle.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-36> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/slider_right_handle.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-37> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/subfolder.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-38> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/title_grip_
   horizontal.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-39> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/title_grip_vertical.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-40> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/titlebar_float.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-41> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/tree17_branch-closed.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-42> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/tree17_branch-
   closed_nosib.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-43> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/tree17_branch-end.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-44> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/tree17_branch-more.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-45> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/tree17_branch-open.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-46> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/tree17_branch-
   open_nosib.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-47> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/tree17_vline.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-48> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/tree_branch-closed.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-49> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/tree_branch-
   closed_nosib.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-50> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/tree_branch-end.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-51> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/tree_branch-more.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-52> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/tree_branch-open.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-53> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/tree_branch-
   open_nosib.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-54> (0)
   - *A* stuff/config/qss/Minimalist_072/imgs/white/tree_vline.png
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-55> (0)
   - *A* stuff/config/qss/Minimalist_072/less/base/colors.less
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-56> (28)
   - *A* stuff/config/qss/Minimalist_072/less/base/mixins.less
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-57> (22)
   - *A* stuff/config/qss/Minimalist_072/less/components/all.less
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-58> (9)
   - *A* stuff/config/qss/Minimalist_072/less/components/boxes.less
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-59> (11)
   - *A* stuff/config/qss/Minimalist_072/less/components/buttons.less
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-60> (90)
   - *A* stuff/config/qss/Minimalist_072/less/components/icons.less
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-61> (34)
   - *A* stuff/config/qss/Minimalist_072/less/components/tabs.less
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-62> (70)
   - *A* stuff/config/qss/Minimalist_072/less/components/treeviews.less
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-63> (94)
   - *A* stuff/config/qss/Minimalist_072/less/layouts/all.less
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-64> (21)
   - *A* stuff/config/qss/Minimalist_072/less/layouts/controls.less
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-65> (197)
   - *A* stuff/config/qss/Minimalist_072/less/layouts/filebrowser.less
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-66> (100)
   - *A* stuff/config/qss/Minimalist_072/less/layouts/filmstrip.less
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-67> (16)
   - *A* stuff/config/qss/Minimalist_072/less/layouts/format.less
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-68> (90)
   - *A* stuff/config/qss/Minimalist_072/less/layouts/mainwindow.less
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-69> (378)
   - *A* stuff/config/qss/Minimalist_072/less/layouts/misc.less
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-70> (46)
   - *A* stuff/config/qss/Minimalist_072/less/layouts/palette.less
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-71> (151)
   - *A* stuff/config/qss/Minimalist_072/less/layouts/popups.less
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-72> (69)
   - *A* stuff/config/qss/Minimalist_072/less/layouts/schematic.less
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-73> (58)
   - *A* stuff/config/qss/Minimalist_072/less/layouts/scriptconsole.less
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-74> (19)
   - *A* stuff/config/qss/Minimalist_072/less/layouts/taskviewer.less
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-75> (30)
   - *A* stuff/config/qss/Minimalist_072/less/layouts/toolbar.less
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-76> (99)
   - *A* stuff/config/qss/Minimalist_072/less/layouts/viewer.less
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-77> (96)
   - *A* stuff/config/qss/Minimalist_072/less/layouts/xsheet.less
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-78> (213)
   - *A* stuff/config/qss/Minimalist_072/less/main.less
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-79> (19)
   - *A* stuff/config/qss/Minimalist_072/less/mods/all.less
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-80> (5)
   - *A* stuff/config/qss/Minimalist_072/less/mods/scrollbars.less
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-81> (49)
   - *A* stuff/config/qss/Minimalist_072/less/themes/Minimalist_072.less
   <https://github.com/opentoonz/opentoonz/pull/1112/files#diff-82> (365)

Patch Links:

   - https://github.com/opentoonz/opentoonz/pull/1112.patch
   - https://github.com/opentoonz/opentoonz/pull/1112.diff

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/opentoonz/opentoonz/pull/1112>, or mute the thread
<https://github.com/notifications/unsubscribe-auth/APkuZAOgUSPSccUNXlZey_B6sl4rFtXAks5rntwugaJpZM4Mi8gS>
.

artisteacher commented:
On a Mac, the size of the lock icons, the vertical text alignment and the clipping on the startup pop-up is fixed.

I hadn't noticed this before, but it is true of the other themes as well, but the viewer buttons overlap the text to the left if the viewer is too narrow. This occurs on the other theme as well. When the buttons do need to overlap in that area, it would be better not to see the text underneath.
![screen shot 2017-03-21 at 7 28 02 am](https://cloud.githubusercontent.com/assets/21159116/24145985/0c4f73a8-0e0a-11e7-9985-8da23ef75f03.png)

In areas with tabs, it would be nice to have a little bit of a margin/padding between the tabs and content.
![screen shot 2017-03-21 at 8 00 48 am](https://cloud.githubusercontent.com/assets/21159116/24146574/a4a78fc6-0e0c-11e7-8629-c6fdf6a5cc9b.png)
![screen shot 2017-03-21 at 8 00 41 am](https://cloud.githubusercontent.com/assets/21159116/24146579/a813c436-0e0c-11e7-9a9e-3c4c27ad31be.png)

Another thing I just found - and is also true of other themes, if the combo viewer is too narrow, there is some unfortunate clipping/overlapping on the left side of the tool option bar
![screen shot 2017-03-21 at 7 50 00 am](https://cloud.githubusercontent.com/assets/21159116/24146306/850ddd88-0e0b-11e7-9024-3a86b4ecb5a3.png)
![screen shot 2017-03-21 at 7 50 35 am](https://cloud.githubusercontent.com/assets/21159116/24146311/8a0f701c-0e0b-11e7-9230-f40bfe02b571.png)

These buttons all share the same function, would it make more sense for them to all have the same style?
From the tool bar:
![screen shot 2017-03-21 at 8 04 21 am](https://cloud.githubusercontent.com/assets/21159116/24146724/519da904-0e0d-11e7-9166-23555d6eb649.png)
From the viewer:
![screen shot 2017-03-21 at 7 49 20 am](https://cloud.githubusercontent.com/assets/21159116/24146371/c6a6d362-0e0b-11e7-8173-e853001283a7.png)
From the level palette:
![screen shot 2017-03-21 at 7 52 30 am](https://cloud.githubusercontent.com/assets/21159116/24146380/d15eca6c-0e0b-11e7-9cb3-42f26d7994e2.png)

In cases where there is overlapping, it might be helpful if the button background was slightly darker or perhaps translucent (color model pictured below).
![screen shot 2017-03-21 at 8 15 00 am](https://cloud.githubusercontent.com/assets/21159116/24146991/9d5764ce-0e0e-11e7-84ce-6a5766d29525.png)

I notice that the tool option bar is inside of the viewer in your preview image - like it is in the combo-viewer. I'm not sure if that is something you have customized, or something that works differently on a mac. On my computer, the tool option bar is outside of the normal viewer which makes for a very visually crowded menu area.
![screen shot 2017-03-21 at 8 08 54 am](https://cloud.githubusercontent.com/assets/21159116/24146852/fd6dd7fe-0e0d-11e7-86b5-eb0fb9e15e85.png)

I'm not the biggest fan of the white lines - they are distracting on the dark themes.
![screen shot 2017-03-21 at 8 17 49 am](https://cloud.githubusercontent.com/assets/21159116/24147083/092dfeba-0e0f-11e7-9cab-5e22472646f2.png)

@konero, you have made some great progress. It's fantastic that you got this out so quickly. I'm happy to give feedback any time!

turtletooth commented:
@konero - If we go with this style - fixing the pair value is a pretty easy fix -
get rid of all the other painter.draw stuff and replace it with:
p.fillRect(x0, y - 2, x1 - x0, 4, getDarkLineColor());

However doing that would break the current themes.  How hard would it be to make a quick light and darker version of this to replace the existing themes?

BTW- how attached are you to the orange?  I can't decide if the orange and green work well together.

@shun-iwasawa What are your thoughts in replacing the current themes altogether and replacing them with this basic theme?

konero commented:
@turtletooth Not attached to colors at this stage, its all still experimental. I was using orange to find some bugs since its a nice powerful color and change it frequently to avoid fatigue!

![tz-light-prev](https://cloud.githubusercontent.com/assets/19820721/24230819/f9eadb4c-0f78-11e7-8de6-473ba132d4e2.png)

I'm still trying out icon variants, I thought the current lock icons I made were a bit strong and not so obvious so I've redone those.

Feel free to suggest colors, you can play with the selector color by editing `less/themes/Minimalist_072.less` > `@selected-bg-color`. All colors for the theme are controlled in that file.

artisteacher commented:
Another small spacing issue in the style editor - visually it would be more consistent with the rest of the editor if there was a small margin to the left of the color box and the text directly above.
![screen shot 2017-03-23 at 8 56 26 am](https://cloud.githubusercontent.com/assets/21159116/24254579/ae3a2734-0fb9-11e7-8aba-af370a2751e2.png)

 I hope the resulting theme(s) end up as a fusion between this and @turtletooth's dark theme. Both have some great features.

konero commented:
@artisteacher 

> I hope the resulting theme(s) end up as a fusion between this and @turtletooth's dark theme.

What parts are you interested in fusing?

I've been trialing a more flat design, which is where buttons have no borders etc. Everyone seems to request it, "make it more like Maya/TVPaint" etc, I've resisted it, but I'm starting to trial it now. What do you think? Go flat? 🤔 

I've started work on 048 (Dark), some work still to do (don't report bugs on this yet, its very rough)
**DL: [Minimalist_048.zip](https://github.com/opentoonz/opentoonz/files/866190/Minimalist_048.zip)**

072 (Flat ver)
**DL: [Minimalist_072.zip](https://github.com/opentoonz/opentoonz/files/866191/Minimalist_072.zip)**

![072-flat-01](https://cloud.githubusercontent.com/assets/19820721/24270333/a89e4a1e-100c-11e7-965a-bd473e7a8cbc.png)![048-flat-01](https://cloud.githubusercontent.com/assets/19820721/24270406/f132ab8a-100c-11e7-8688-c2d809455357.png)

turtletooth commented:
@konero In general, I really like what you have going on.  It's very clean.  I really like what you did with tabs.

I'm not sure what I think of the borderless dropdowns and buttons.  Sometimes it seems a little close to how I would expect a disabled item to appear:

![active](https://cloud.githubusercontent.com/assets/4576381/24274505/3419554e-0fee-11e7-90fb-cd0bc2daf060.PNG)

There are some elements of the Astral theme that I miss:

- I liked the lock icons (were they the Google material icons?)
- I preferred the checkbox size from Astral 
- I liked the preferences sidebar contrast in Astral
- I miss Segoe-UI (I know that needs to be the way it is right now)
- I think I prefer the active tools and viewer controls having a dark background instead of the highlighted color.

What was it about Astral that forced you to switch to the new theme?  

You are doing great things here.  Thank a ton!

konero commented:
@turtletooth Thanks! I reworked it a bit from Astral for Linux support, I'm still not entirely sure why but some parts didn't work on Linux, probably due to the font issues over-expanding their containers.

I miss Segoe UI too, it just doesn't look like a Windows app without it. Do you think you could give `main.cpp` a look and see how fonts are specified with `qfont`? It would be great to nip this problem in the bud once and for all.

My problem with it is Linux and Windows share the same QSS file, so if something doesn't work on one a compromise has to be made on the other. So far, it seems to just be font issues.

I can only guess, but do you reckon Linux needs an `ifdef LINUX` and its own font-family? Sorry, as I say I'm too rusty with C++, it's not something I can look into.

This may be a good example: https://doc.qt.io/archives/3.3/simple-font-demo-example.html

Docs:
http://doc.qt.io/qt-5/qfontinfo.html#family
http://doc.qt.io/qt-5/qfont.html#family
http://doc.qt.io/qt-5/qfont.html#fontmatching

What we want is 3 separate font-family stacks that are loaded depending on what OS is being used. We especially don't want it to be bold either if we want to read it in Roman letters.

**Win10, Win 8.1, Win 8, Win 7, Win Vista, Win XP**
```
#ifdef WIN32
  Segoe UI, Tahoma, sans-serif
#endif
```
**El Capitan, Yosemite, Mavericks**
```
#ifdef MAC (or __APPLE__, IDK, whatever the compiler wants)
  San Francisco, Helvetica Neue, Lucida Grande, Helvetica
#endif
```
**Ubuntu, Mint**
```
#ifdef LINUX
  Ubuntu, Sans
#endif
```
You can probably find it by doing a search for "Arial", or find that typo string "Hervetica".

If @artisteacher wants to confirm if changes work on Mac, we could update it at the source.

I'm also in the same boat as you about border-less controls, I haven't used Maya or TVPaint 11 (I'm still on 10.5.7) so it may be my bias.

> I liked the lock icons (were they the Google material icons?)

No, I didn't think to use Material, I'll give them a go. 👍 

> I liked the preferences sidebar contrast in Astral

No prob, I'll add it back, it was due to the bug you reported above.

> I preferred the checkbox size from Astral

They're still the same size from what I can see, maybe the light border makes them look smaller?

> I think I prefer the active tools and viewer controls having a dark background instead of the highlighted color.

Me too. I was trying to see what the Maya style would look like and seeing it if would grow on me.

konero commented:
@turtletooth & @artisteacher 

I've re-worked the Play Bar and the scrollable toolbar buttons again. This was NOT easy. 😭 

![tnz-new-playbar](https://cloud.githubusercontent.com/assets/19820721/24287212/ad9386ec-1070-11e7-933f-ed409fe88579.gif)

tGuMedia commented:
@konero it's looking very good, may i ask if there's going to be a Minimalist_128?

FixAndersen commented:
Please create adobe After Effects interface. 
![image](https://cloud.githubusercontent.com/assets/26668270/24320453/c59d38f6-1146-11e7-9361-037989acc2d1.png) 
![image](https://cloud.githubusercontent.com/assets/26668270/24320465/f6ff3322-1146-11e7-94ec-ff4f001c0b24.png) 
![image](https://cloud.githubusercontent.com/assets/26668270/24320490/590d94a0-1147-11e7-8f41-b9654bb12329.png) 
![image](https://cloud.githubusercontent.com/assets/26668270/24320500/98d2d398-1147-11e7-8081-56f6d2cac48e.png) 
![image](https://cloud.githubusercontent.com/assets/26668270/24320501/a5995840-1147-11e7-8f20-f7f6707f8ac4.png)

konero commented:
@FixAndersen I hear you, but copying another vendors UI is not good practice. It's only my intention to clean up the current UI with an alternate theme based on various feedback sources.

Back to the console bar again, this time I tried to tackle the FPS slider. I tried to make it "responsive", so when the window is too small for the slider to have any real world usefulness it should vanish and fall back to the arrow buttons.

Seems to work OK.

![responsive-fps-slider](https://cloud.githubusercontent.com/assets/19820721/24321787/f7f8b11c-114c-11e7-93ee-643c6262199e.gif)

UpgradeOpenToonz commented:
When should we wait for this? ![image](https://cloud.githubusercontent.com/assets/26676791/24324726/86e74326-119c-11e7-81b8-13a1c675a291.png)
![image](https://cloud.githubusercontent.com/assets/26676791/24324732/a41ba180-119c-11e7-8886-db872514d0dc.png)

AnimatorGF commented:
Amazing @UpgradeOpenToonz

konero commented:
@UpgradeOpenToonz - That's a concept I did with somebody a long time ago, this PR is the result of that but @turtletooth referenced it for his theme in #1043. If you're specifically referring to the icons in the options bar - we tried it but it wasn't worth the hassle since each of those functions already apply to shortcut keys which is a lot quicker to use when you're holding a stylus. If anything, in hindsight those icons would probably produce bad habits.

artisteacher commented:
So I tried out the most recent versions, but they do not seem to be working properly - they both look the same.
![screen shot 2017-03-28 at 8 28 56 am](https://cloud.githubusercontent.com/assets/21159116/24404827/b434033c-1390-11e7-9dd9-647c4a1f6892.png)
In terms of the dark theme, I really like how the style editor is handled - there is the right amount of margin to make the tabs and buttons very clear.
![screen shot 2017-03-28 at 8 30 24 am](https://cloud.githubusercontent.com/assets/21159116/24404964/1c29a35c-1391-11e7-973f-b0b305cdba5e.png)

The new scrub bar and fps slider both look great! I agree with turtletooth about having borders around controls.

konero commented:
@artisteacher That's odd, could you download the file in the OP and give that one a try? I updated the ZIP there with the most recent version (I wasn't sure if you downloaded them from other posts, those are very patchy, the one in the OP should be stable).

In regards to the Style Editor channel buttons I couldn't agree more, its a shame they have no objectNames then I could turn them into pill tabs or something. But for now I've done as you suggested and added some margin.

![style-ed-upd](https://cloud.githubusercontent.com/assets/19820721/24409236/89ac449e-13c7-11e7-84a0-03b71c3f39be.png)

Thanks for the feedback! 👍

artisteacher commented:
Looking good! A couple small notes:

 At least on my screen, in 048, the active room tab state is too similar to the inactive state (especially with the viewer or etc. expanded). As a side note - I really like tabs with slightly rounded corners at the top.
![screen shot 2017-03-29 at 1 00 21 pm](https://cloud.githubusercontent.com/assets/21159116/24466701/c6634a28-147f-11e7-8c6c-7851a88c121b.png)
On 128, the viewer rulers and default background color are too similar.
![screen shot 2017-03-29 at 1 07 05 pm](https://cloud.githubusercontent.com/assets/21159116/24467036/eec23366-1480-11e7-9dce-50495751e5e4.png)
Lastly, this is in all of the themes, but should this xsheet button be overlapping with the columns?
![screen shot 2017-03-29 at 1 14 52 pm](https://cloud.githubusercontent.com/assets/21159116/24467317/06143ea0-1482-11e7-8af4-90e7c3ad998b.png)

![screen shot 2017-03-29 at 1 14 59 pm](https://cloud.githubusercontent.com/assets/21159116/24467300/fa14243a-1481-11e7-8e27-5c2c4b092587.png)


I also ran into a bug? where the room tabs did not update when the theme changed. Changing rooms resolved the issue and now the themes all load as expected every time. Not sure what changed there...

turtletooth commented:
@konero You are awesome!
One last thought- What about changing the names for the three versions to something more descriptive?  Minimalist_048 doesn't help the user know what they are getting?  

But WOW, do I love the work you are doing here!

turtletooth commented:
@artisteacher 
I think the xsheet pull down may need to be fixed in code.  It seems to be that way on all themes.

I actually like the tabs the way @konero has them now.  I feel like it's a cleaner look.

artisteacher commented:
@turtletooth, the tabs are cleaner - I'm just biased towards subtle rounded corners. I feel like they always read more as tabs. It's not important though - just a personal preference. Either way, these themes have made a lot of progress!

With 048, if another window is underneath the level strip, it can be tricky finding the divider to adjust the size - due to similarities in value. I really like the super dark dividers though and have not noticed an issue in other configurations.
![screen shot 2017-03-29 at 3 03 39 pm](https://cloud.githubusercontent.com/assets/21159116/24471856/775fcc5a-1491-11e7-8093-fe016c8562f5.png)
The inactive channel buttons do not necessarily read as inactive. Perhaps having the text color a little closer in value to the button background would help with that.

konero commented:
@artisteacher Thanks for the feedback!

That bug may be an issue with the Mac build, I'm always reading about quirky issues like that on Mac, I haven't been able to reproduce anything on my end (Win 10).

Good catch on the 128 ruler color, I've fixed it.

As @turtletooth describes looks like the XSheet combo box is misplaced (perhaps the whole surrounding container too) in the code. 

I always have my FilmStrip docked down the side (from top to bottom of screen) so I didn't run into that issue. I don't think there's much we can do there though; the price of minimalism I guess.

> The inactive channel buttons do not necessarily read as inactive. Perhaps having the text color a little closer in value to the button background would help with that.

For the 048 channel buttons I'm not sure how to tackle them, any change made to them changes all buttons in the same window which can be rather dirty, its better to wait for improvement in the code just in case new features add more buttons such as #1124.

For the room tabs, I didn't really test on conventional displays, everything we use here is professionally calibrated, I've adjusted the contrast a bit, I hope that fixes it. I used an app called Flix before which is what I based the room tabs from (they have theirs in the same place), I really liked the cleanness of them, and the ones in VSCode are similar, which is what I've been using to build the themes.

![flix-tab-tz](https://cloud.githubusercontent.com/assets/19820721/24497190/81313222-1532-11e7-838f-0bd9d444f07f.png)

konero commented:
@turtletooth Thanks for the kind words!

Sure we can change the names, I'm not really sure what to name them though. 👍

turtletooth commented:
@konero Thanks for the updates.  Sorry I haven't been around- we've moved and are trying to get our lives organized.  

It looks like we may need to ditch the colored background in settings.  I am getting the weird glitch again:

![settings](https://cloud.githubusercontent.com/assets/4576381/24729213/5acaa7a2-1a1a-11e7-8651-d0761b73311d.PNG)

As far as names go, what do you think about neutral, dark and darker (without the numbers)?  

I think these are to the point where they can replace the current themes entirely.  What are your thoughts @shun-iwasawa ?

konero commented:
@turtletooth I hope things are going well!

My workstation is all over my desk, since I'm trialing Ryzen. But I tried cloning my repo to another machine but it failed on lfs, looks like AppVeyor is having the same problem?

`fatal: thirdparty/LibJPEG/jpeg-9/lib/LibJPEG-9_2013_64.lib: smudge filter lfs failed`

I got it to work by using git clone `https://konero@github.com/konero/opentoonz.git`, is it a problem on my end?

I haven't bumped into that bug yet on Windows, I'll change the config again in any case.

I'm hesitant on the names. I think @shun-iwasawa said something about the colors having a purpose, so I wonder if users choose the theme based on the rgb value for a specific reason. For example, 128 is completely neutral/middle-gray in sRGB, its like using a gray card in photography. It's also what the default Viewer BG Color is set to.

You can check the table here: https://en.wikipedia.org/wiki/Middle_gray

![gc2](https://cloud.githubusercontent.com/assets/19820721/24754143/2387b4de-1acd-11e7-8ee2-4ab2b61cfb53.PNG)

For now I've changed it back to what Astral was using plus the descriptors. Someone with more insight can probably change them if they're no good.

konero commented:
OK. Hopefully I've fixed that prefs issue now. I added a larger font-size which seems to do the trick. It's ugly, but I think whats causing it is by me trying to remove the indent that happens on hover state.

I managed to reproduce it, and this is what fixed it for me.

damies13 commented:
Hi All,

Actually I was only having that issue with AppVeyor build server, on my
machines I was not having any issues with git activities (clone / pull /
etc) could it be a regional issue? I don't know where AppVeyor build server
is, FYI I am in Brisbane, Australia if that helps.

Dave.



On 7 April 2017 at 00:43, Kite <notifications@github.com> wrote:

> OK. Hopefully I've fixed that prefs issue now. I added a larger font-size
> which seems to do the trick. It's ugly, but I think whats causing it is by
> me trying to remove the indent that happens on hover state.
>
> I managed to reproduce it, and this is what fixed it for me.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/opentoonz/opentoonz/pull/1112#issuecomment-292196621>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AOhvZ1WfIc-yUaVfGXj0BeD76GhEI3n8ks5rtPohgaJpZM4Mi8gS>
> .
>

damies13 commented:
@konero, try your build again, it seems the issue with AppVeyor's build server has been resolved (my build succeeded last night). All I did was close and re-open the build request.

Good luck,

Dave.

konero commented:
Quick note, don't review/merge yet. I want to change some things, like the convoluted scrollbars and work on 128 some more since I've got a lot of feedback for that one, I just need a day or two. I also really want to redo the camera/output icons.

konero commented:
I'd say this is ready for review now.

The last commit has a lot of changes.

I changed 128 to skeuomorphic, its a lot easier to read at this color range and hopefully deals with some of the issues raised by @artisteacher in the style editor.

![skeuo-128](https://cloud.githubusercontent.com/assets/19820721/24840482/db1f5ddc-1d65-11e7-82f0-651360aec86f.png)

I also changed the scroll bar design, I actually prefer the previous design -but- this new one is almost set and forget developer wise, the last one was convoluted, almost every widget needed tweaking to adapt to it, so I'll stick with the new one, which is based on Windows 10 scroll bar design.

I'll update the OP with latest files/previews.

konero commented:
**Note:** Don't merge, trialing SVG again. Needs OSX confirmation.

I attempted SVG icons again for the Output/Cleanup settings box, they didn't work last time on OSX. If this works I can make SVG icons for all other icons.

I confirm they work fine on Linux (Ubuntu) and Windows 10. I don't have a Mac, neither do I know anybody with one. It would be appreciated if a Mac user could give this a test run, perhaps @artisteacher or @shun-iwasawa?

**[Astral_072_Dark.zip](https://github.com/opentoonz/opentoonz/files/910177/Astral_072_Dark.zip)**

They should look like this image and not squashed.

![svg-test-prev](https://cloud.githubusercontent.com/assets/19820721/24864064/fc28bc18-1dfa-11e7-9768-3085fd2057f1.PNG)

artisteacher commented:
The size looks fine - not squished or tiny. The locked and unlocked states look the same. I did notice that there is an offset border when selecting the buttons.
![screen shot 2017-04-10 at 10 55 51 am](https://cloud.githubusercontent.com/assets/21159116/24867865/840eb750-1ddc-11e7-84e5-91565c80108e.png)
![screen shot 2017-04-10 at 10 59 30 am](https://cloud.githubusercontent.com/assets/21159116/24868023/dc088f08-1ddc-11e7-9925-4e14894c05ec.png)
With the forced square pixel button, the size fits so the outline is not super obvious.
![screen shot 2017-04-10 at 10 59 44 am](https://cloud.githubusercontent.com/assets/21159116/24868049/eca270b8-1ddc-11e7-8dc9-3486c3243325.png)

turtletooth commented:
@shun-iwasawa With these themes, do you think there is a need for the old themes anymore?

shun-iwasawa commented:
@konero Thank you for the great work. I'll review this and will ask my acquaintance look and feel.

@turtletooth Basically I agree with you that it can replace the old theme. At least it can be the default. Still not sure we can delete the old themes now or in the next version.

tGuMedia commented:
@shun-iwasawa i don't think the old themes should be deleted.

konero commented:
@shun-iwasawa Thanks.

If you'd prefer, I can always do the typical 3D look like Clip Studio etc. I know "flatty" UI isn't for everybody.

![skeuo-test-72-2](https://cloud.githubusercontent.com/assets/19820721/24918441/575bda9c-1ed8-11e7-9596-954f8c651618.png)

blurymind commented:
Please add flatty as an option. I will use it for sure!

On 11 Apr 2017 17:01, "Kite" <notifications@github.com> wrote:

> @shun-iwasawa <https://github.com/shun-iwasawa> Thanks.
>
> If you'd prefer, I can always do the typical 3D look like Clip Studio etc.
> I know "flatty" UI isn't for everybody.
>
> [image: skeuo-test-72-2]
> <https://cloud.githubusercontent.com/assets/19820721/24918441/575bda9c-1ed8-11e7-9596-954f8c651618.png>
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/opentoonz/opentoonz/pull/1112#issuecomment-293310920>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGMbVT3FPLadMpkxsxUjxQRYA9__F8u0ks5ru6PegaJpZM4Mi8gS>
> .
>

AyameLee commented:
+1 for flat!

shun-iwasawa commented:
Confirmed that it works fine on Windows and SVG images looks so nice on high DPI monitor.
Thanks again @konero for this PR!

- For now the `Preferences>Interface>style` is saved in the file NOT as the "selected style name", but as the "selected index of the combo box". It will cause the current style to  be unexpectedly changed when merging a PR which adds a new style ( like this PR ). I will modify it soon to save the style name instead of the index.

- Based on the @tGuMedia opinion, I think it would better to leave the old styles for a while. I would like to evaluate how troublesome to maintain both old&new styles when implementing a new gadgets to the software.

- @konero it would be much appreciated if you are interested in converting (or even re-designing) `Resources` PNG images to SVG as well. Actually the toolbar icons are already SVGs and can be smoothed in high dpi monitors by just adding `QCoreApplication::setAttribute( Qt::AA_UseHighDpiPixmaps );` I'll try introducing it.

- IMO it would be helpful for other designers if @konero 's description in the first post of this conversation is included in the repository.

I'll merge this tomorrow, as soon as I check the OSX build. Thank you for your patience!

konero commented:
@shun-iwasawa I see, if that is the case what do you think about making the 072 variant default? That would put OpenToonz in the same common denominator as TVPaint, Krita, Maya, Photoshop, DaVinci, Clip Studio etc.

I agree with keeping the original themes, I admit there may be a small dev learning curve with this theme, best to keep a safe and familiar fallback (that goes for end-users too).

Sure I can work on the resource icons converting them to SVG, I'll tackle all the front-facing icons I can see in the program. 👍

shun-iwasawa commented:
Confirmed that it works fine on OSX.
According to [this page](http://blog.qt.io/blog/2013/04/25/retina-display-support-for-mac-os-ios-and-x11/) you have to set some keys to `Info.Plist` in order to get smooth image on retina display.
I'll try to find a way to set them automatically.

LGTM
